bindings/java/src/main/com/apple/foundationdb/FDBTenant.java

```diff
@@ -198,6 +200,19 @@ public CompletableFuture<Long> verifyBlobRange(byte[] beginKey, byte[] endKey, l
 		}
 	}
 
+	@Override
+	public CompletableFuture<Long> getId(Executor e) {
+		if (this.idFuture == null) {
```

Comment:
I think this has a possible race condition if you were to simultaneously call this from two threads. You could move it all within the lock scope, or potentially you could try to do some double-checked locking.